accounts-db/src/append_vec.rs

```diff
@@ -259,18 +291,23 @@ impl Drop for AppendVec {
     fn drop(&mut self) {
         APPEND_VEC_MMAPPED_FILES_OPEN.fetch_sub(1, Ordering::Relaxed);
         match &self.backing {
-            AppendVecFileBacking::MmapOnly(mmap_only) => {
+            AppendVecFileBacking::Mmap(mmap_only) => {
                 if mmap_only.is_dirty.load(Ordering::Acquire) {
                     APPEND_VEC_MMAPPED_FILES_DIRTY.fetch_sub(1, Ordering::Relaxed);
                 }
             }
+            AppendVecFileBacking::File(_) => {}
         }
-        if let Err(_err) = remove_file(&self.path) {
-            // promote this to panic soon.
-            // disabled due to many false positive warnings while running tests.
-            // blocked by rpc's upgrade to jsonrpc v17
-            //error!("AppendVec failed to remove {}: {err}", &self.path.display());
-            inc_new_counter_info!("append_vec_drop_fail", 1);
+        if self.remove_file_on_drop.load(Ordering::Acquire) {
```

Comment:
Would it be possible for the FileBacking account storage to be removed by account_storage map, in case like purge_dead_slots, or shrinking? 

My concern is that `remove_file_on_drop` will always be false. When we replace mmap-appendvec with file-based, and inserted it into account_storage map. And later when the slot is dropped from the map, we won't be deleting the backing file?